app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -531,6 +540,50 @@ class TalkTopicActivity : BaseActivity(), LinkPreviewDialog.Callback {
         }
     }
 
+    override fun onUserMentionListUpdate() {
+        binding.talkScrollContainer.post {
+            if (!isDestroyed && !userMentionScrolled) {
+                binding.talkScrollContainer.smoothScrollTo(0, binding.talkScrollContainer.height * 2)
+                userMentionScrolled = true
+            }
+        }
+    }
+
+    override fun onUserMentionComplete() {
+        userMentionScrolled = false
+    }
+
+    private fun parseUserNamesFromTopic(): Set<String> {
+        val userNames = mutableSetOf<String>()
+        // Go through our list of replies under the current topic, and collect any links to user
+        // names, making sure to store them in reverse order, so that the last user name mentioned
+        // in a response will appear first in the list of hints when searching for mentions.
+        topic?.replies?.forEach {
+            var start = 0
+            val userList = mutableListOf<String>()
+            while (true) {
+                val searchStr = "title=\""
+                start = it.html!!.indexOf(searchStr, startIndex = start)
+                if (start < 0) {
+                    break
+                }
+                start += searchStr.length
+                val end = it.html!!.indexOf("\"", startIndex = start)
+                if (end <= start) {
+                    break
+                }
+                val name = it.html!!.substring(start, end)
+                val title = PageTitle(name, pageTitle.wikiSite)
+                if (title.namespace() == Namespace.USER || title.namespace() == Namespace.USER_TALK) {
+                    userList.add(0, StringUtil.removeUnderscores(title.text))
+                }
+                start = end
+            }
+            userList.forEach { userNames.add(it) }
```

Comment:
Can we use `userNames.addAll(userList)`?

Reply:
yes!